crates/bevy_math/src/cubic_splines.rs

```diff
@@ -279,9 +290,269 @@ impl<P: Point> CubicGenerator<P> for CubicBSpline<P> {
         let segments = self
             .control_points
             .windows(4)
-            .map(|p| CubicCurve::coefficients([p[0], p[1], p[2], p[3]], 1.0 / 6.0, char_matrix))
+            .map(|p| CubicSegment::coefficients([p[0], p[1], p[2], p[3]], 1.0 / 6.0, char_matrix))
+            .collect();
+
+        CubicCurve { segments }
+    }
+}
+
+/// Error during construction of [`CubicNurbs`]
+#[derive(Debug, Error)]
+pub enum CubicNurbsError {
+    /// Provided knot vector had an invalid length.
+    #[error("Invalid knot vector length: expected {expected}, provided {provided}")]
+    InvalidKnotVectorLength {
+        /// Expected knot vector length
+        expected: usize,
+        /// Provided knot vector length
+        provided: usize,
+    },
+    /// Knot vector has invalid values. Values of a knot vector must be nondescending, meaning the
+    /// next element must be greater than or equal to the previous one.
+    #[error("Invalid knot vector values: elements are not nondescending")]
+    InvalidKnotVectorValues,
+    /// Provided weights vector didn't have the same amount of values as the control points vector.
+    #[error("Invalid weights vector length: expected {expected}, provided {provided}")]
+    WeightsVectorMismatch {
+        /// Expected weights vector length
+        expected: usize,
+        /// Provided weights vector length
+        provided: usize,
+    },
+    /// The amount of control points provided is less than 4.
+    #[error("Not enough control points, at least 4 are required, {provided} were provided")]
+    NotEnoughControlPoints {
+        /// The amount of control points provided
+        provided: usize,
+    },
+}
+
+/// A spline interpolated continuously across the nearest four control points.
+///
+/// ### Interpolation
+/// The curve does not pass through control points, unless the knot vector has 4 consecutive values
+/// that are equal to each other
+///
+/// ### Tangency
+/// Automatically computed based on the position of control points.
+///
+/// ### Continuity
+/// C2 continuous! The acceleration continuity of this spline makes it useful for camera paths.
```

Comment:
Amazing work!

I think it would be worth doing another pass on this particular bit of documentation. Here is something basic, I imagine others can do better.

```suggestion
/// A cubic non-uniform rational B-spline (NURBS). Generates a smooth curve from a 
/// sequence of control points by interpolating between four points at a time.
///
/// ### Interpolation
/// The knot vector is a non-decreasing sequence that controls which four
/// control points are assigned to each segment of the curve. It can be used to make 
/// sharp corners. The curve will not pass through the control points unless the 
/// knot vector has the same value four times in a row.
///
/// ### Curvature
/// The tangents automatically calculated based on the position of the control
/// points. The curve is C2 continuous (meaning both the velocity and
/// acceleration are smooth), making it useful for camera paths and moving objects.
```

Not an expert, but I was wondering if the curve really is always C2 when the knot vector is nonuniform.

Reply:
Continuity would decrease only when there are knots with the same value. Multiplicity of 2 decreases it to C1, multiplicity of 3 to C0.

Reply:
Adjusted the documentation to include more information